tensorboardX/writer.py

```diff
@@ -239,18 +238,60 @@ def __init__(self, log_dir=None, comment=''):
             v *= 1.1
         self.default_bins = neg_buckets[::-1] + [0] + buckets
         self.text_tags = []
+        #
+        self.all_writers = {self.file_writer.get_logdir() : self.file_writer}
+        self.scalar_dict = {} # {writer_id : [[timestamp, step, value],...],...}
+
+
+    def __append_to_scalar_dict(self, tag, scalar_value, global_step,
+                                timestamp):
+        """This adds an entry to the self.scalar_dict datastructure with format
+        {writer_id : [[timestamp, step, value], ...], ...}.
+        """
+        if not tag in self.scalar_dict.keys():
+            self.scalar_dict[tag] = []
+        self.scalar_dict[tag].append([timestamp, global_step, scalar_value])
+ 
     def add_scalar(self, tag, scalar_value, global_step=None):
         """Add scalar data to summary.
-
         Args:
             tag (string): Data identifier
             scalar_value (float): Value to save
             global_step (int): Global step value to record
 
         """
-        scalar_value = makenp(scalar_value)
-        assert(scalar_value.squeeze().ndim==0), 'input of add_scalar should be 0D'
-        self.file_writer.add_summary(scalar(tag, scalar_value), global_step)
+        np_scalar_value = makenp(scalar_value)
+        assert(np_scalar_value.squeeze().ndim==0), 'input of add_scalar should be 0D'
+        self.file_writer.add_summary(scalar(tag, np_scalar_value), global_step)
+        self.__append_to_scalar_dict(tag, scalar_value, global_step, time.time())
```

Comment:
demo.py breaks here. (non-float object might be passed therefore `json.dump fails`